apps/wallet-mobile/src/features/Receive/common/useMultipleAddressesInfo.tsx

```diff
@@ -0,0 +1,52 @@
+import {isBoolean, parseSafe, useAsyncStorage, useMutationWithInvalidations} from '@yoroi/common'
+import {UseMutationOptions, useQuery, UseQueryOptions} from 'react-query'
+
+import {useSelectedWallet} from '../../../SelectedWallet'
+
+const isShowingMultipleAddressInfoKey = 'isShowingMultipleAddressesModal'
+
+const useSetShowMultipleAddressessInfo = (options?: UseMutationOptions<void, Error, boolean>) => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const walletStorage = storage.join(`wallet/${wallet.id}/`)
+
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (show) => walletStorage.setItem(isShowingMultipleAddressInfoKey, show),
+    invalidateQueries: [[wallet.id, isShowingMultipleAddressInfoKey]],
+    ...options,
+  })
+
+  return mutation.mutate
+}
+
+const useIsShowingMultipleAddressesInfo = (options?: UseQueryOptions<boolean, Error, boolean>) => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const walletStorage = storage.join(`wallet/${wallet.id}/`)
+
+  const query = useQuery({
+    suspense: true,
+    queryKey: [wallet.id, isShowingMultipleAddressInfoKey],
+    ...options,
+    queryFn: async () => {
+      const storedStorage = await walletStorage.getItem(isShowingMultipleAddressInfoKey)
+
+      const parsed = parseSafe(storedStorage)
+      // old wallets wont have this key, so we default to true
+      return isBoolean(parsed) ? parsed : true
```

Comment:
marking to display by default (empty storage), it affects only multiple funnel so needs to be configured, looks right needs confirmation cc: @SorinC6